server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -307,6 +311,8 @@ private RolloverResult rolloverDataStream(
             )
             .build();
 
+        newState = writeLoadForecaster.withWriteLoadForecastForWriteIndex(dataStreamName, newState);
```

Comment:
Could we make this an operator `IndexMetadata.Builder -> IndexMetadata.Builder` instead of `ClusterState -> ClusterState` to avoid having to build a new `Metadata` twice?